frontend/packages/dev-console/src/components/import/image-search/ImageStreamTagDropdown.tsx

```diff
@@ -62,18 +71,36 @@ const ImageStreamTagDropdown: React.FC = () => {
     [
       setFieldValue,
       imageStream.image,
-      imageStream.namespace,
       formType,
       application.selectedKey,
       application.name,
       setValidated,
+      imageStream.namespace,
+      initialValues.route.targetPort,
+      touched.imageStream,
     ],
   );
 
   React.useEffect(() => {
     imageStream.tag && searchImageTag(imageStream.tag);
   }, [imageStream.tag, searchImageTag]);
 
+  React.useEffect(() => {
+    if (
+      getIn(touched.imageStream, 'image') &&
```

Comment:
While changing the imagestream if there are more than one ports and user changes target port explicitly in the route advanced options, the target port is reset to the first port in the list of available ports.

Reply:
Maybe add another check for `targetPort` not being touched by user. Then only it should update the value.